src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        copyState = {};
+        break;
+
+      case 'addProperties':
+        copyState = { ...copyState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete copyState[keyToRemove];
+        }
+        break;
+    }
+    states.push({ ...copyState }); // Копируем объект, чтобы избежать мутаций.
```

Comment:
The comment is in a different language (Russian). Consider translating it to English for consistency and clarity, especially if this code is part of a project where English is the primary language.